store/src/directory.rs

```diff
@@ -82,22 +86,129 @@ fn ttl_from_disk(ttl: &[u8]) -> Result<SystemTime, StoreError> {
+            Ok(v) => v,
+        };
+        let mut results = Vec::new();
+        for entry in dir_entries {
+            let entry = match entry {
+                Ok(v) => v,
+                Err(e) => {
+                    log::trace!("Error during maintenance: unable to process entry: {:?}", e);
+                    continue;
+                }
+            };
+            let path = entry.path();
+            match entry.file_type() {
+                Err(e) => {
+                    log::trace!(
+                        "Error during maintenance: Unable to determine file type of {}: {:?}",
+                        path.display(),
+                        e
+                    );
+                    continue;
+                }
+                Ok(v) if v.is_file() => {}
+                Ok(_) => continue,
+            }
+            for neq in &self.neqs {
+                let (key, expected) = neq;
+                match xattr::get(path.clone(), key) {
+                    Ok(Some(v)) => {
+                        let matching = expected.iter().zip(&v).filter(|&(a, b)| a == b).count();
+                        if expected.len() != matching {
+                            results.push(path.clone());
+                        }
+                    }
+                    Ok(None) => {
+                        results.push(path.clone());
+                    }
+                    Err(e) => {
+                        log::trace!("Error checking {}: {}", key, e.to_string());
+                        continue;
+                    }
+                }
+            }
+            for lt in &self.lts {
+                let (key, max) = lt;
+                match xattr::get(path.clone(), key) {
+                    Ok(Some(v)) => {
+                        let value = i64::from_le_bytes(v.try_into().unwrap());
+                        if *max > value {
+                            results.push(path.clone());
+                        }
+                    }
+                    Ok(None) => {}
+                    Err(e) => {
+                        log::trace!("Error checking {}: {}", key, e.to_string());
+                        continue;
+                    }
+                }
+            }
+        }
+        let mut values = Vec::new();
+        for r in results {
+            println!("{:?}", r);
+            let file = match File::open(&r) {
+                Err(e) if e.kind() == std::io::ErrorKind::NotFound => continue,
+                Err(e) => {
+                    log::trace!("Error opening file {}", e.to_string());
+                    continue;
+                }
+                Ok(f) => f,
+            };
+            let val = V::deserialize_from_reader(&file);
+            if let Ok(v) = val {
+                values.push(v)
+            } else {
+                log::trace!("Error deserializing data {:?}", r);
+            }
+        }
+        Ok(Some(ValueIter {
+            index: 0,
+            values: values,
```

Comment:
This can be just `values,`. Rust is nice that way.

Reply:
yup just changed